server/types/text.go

```diff
@@ -36,6 +36,7 @@ var Text = TextType{}
 type TextType struct{}
 
 var _ DoltgresType = TextType{}
+var _ sql.StringType = TextType{}
```

Comment:
This isn't quite how I think this should be approached. I'm assuming `schemaImpl::getKeyColumnsDescriptor` would fail for all types that return `types.ExtendedTypeSerializedWidth_Unbounded`, as I'm guessing that it's due to the serialization format. With that, not all types are going to be string types. In addition, `sql.StringType` implies a kind of contract that Doltgres string types (and Doltgres types in general) won't necessarily fulfill.

The better fix would be to allow any `ExtendedType` as a `UNIQUE` key, since I'm fairly confident that this issue will crop up for all unbounded types (and `sql.StringType` is a non-starter for custom types anyway).